### PR TITLE
fix: checkout new branch before commit

### PR DESCRIPTION
**Pull request by shanejonas**

### CODE REVIEW

**Review by rekmarks — APPROVED** [review]
In the monorepo version I update the `package.json` files and changelogs before creating a new branch, but I guess the order doesn't matter there.

LGTM!

File: run.sh [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
 git config user.name github-actions
 git config user.email github-actions@github.com
 
+node "${GITHUB_ACTION_PATH}/scripts/update-package-version.js"
+
+git checkout -b "${BRANCH_NAME}"
+
 if ! (git add . && git commit -m "Release ${NEW_VERSION}" && git push);
 then
     echo "No changes"
```
Comment (rekmarks) [inline run.sh:21]: Should we treat this as an error case?

Comment (shanejonas) [inline run.sh:21]:
```suggestion
    echo "No changes"
    exit 1
```

Comment (rekmarks) [reply run.sh:21]: Maybe like this? https://github.com/MetaMask/action-monorepo-release-pr/blob/enable-hotfixes/src/create-release-pr.sh#L23-L29

File: run.sh [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-
 git config user.name github-actions
 git config user.email github-actions@github.com
 
-if ! (git add . && git commit -m "Release ${NEW_VERSION}" && git push);
+node "${GITHUB_ACTION_PATH}/scripts/update-package-version.js"
+
+git checkout -b "${BRANCH_NAME}"
+
+if ! (git add . && git commit -m "${NEW_VERSION}");
```
Comment (shanejonas) [inline run.sh:19]:
```suggestion
if ! (git add . && git commit -m "${NEW_VERSION}" && git push);
```

**Review by rekmarks — APPROVED** [review]
LGTM!

**Pull request merged**